sklearn/metrics/tests/test_pairwise.py

```diff
@@ -394,45 +399,44 @@ def test_paired_distances(metric, func):
 
     S = paired_distances(X, Y, metric=metric)
     S2 = func(X, Y)
-    assert_array_almost_equal(S, S2)
+    assert_allclose(S, S2)
     S3 = func(csr_matrix(X), csr_matrix(Y))
-    assert_array_almost_equal(S, S3)
+    assert_allclose(S, S3)
     if metric in PAIRWISE_DISTANCE_FUNCTIONS:
         # Check the pairwise_distances implementation
         # gives the same value
         distances = PAIRWISE_DISTANCE_FUNCTIONS[metric](X, Y)
         distances = np.diag(distances)
-        assert_array_almost_equal(distances, S)
+        assert_allclose(distances, S)
 
 
-def test_paired_distances_callable():
+def test_paired_distances_callable(global_dtype):
     # Test the pairwise_distance helper function
     # with the callable implementation
     rng = np.random.RandomState(0)
     # Euclidean distance should be equivalent to calling the function.
-    X = rng.random_sample((5, 4))
+    X = rng.random_sample((5, 4)).astype(global_dtype, copy=False)
     # Euclidean distance, with Y != X.
-    Y = rng.random_sample((5, 4))
+    Y = rng.random_sample((5, 4)).astype(global_dtype, copy=False)
 
     S = paired_distances(X, Y, metric="manhattan")
     S2 = paired_distances(X, Y, metric=lambda x, y: np.abs(x - y).sum(axis=0))
-    assert_array_almost_equal(S, S2)
+    assert_allclose(S, S2)
```

Comment:
It could be worth to check the output `dtype` here.

Reply:
it does not preserve dtype because manhattan distances doesn't yet